lib/protocol/sequences/Query.js

```diff
@@ -24,12 +24,29 @@ function Query(options, callback) {
   this._fields    = [];
   this._index     = 0;
   this._loadError = null;
+  this._started = false;
 }
 
 Query.prototype.start = function() {
+  this._started = true;
   this.emit('packet', new Packets.ComQueryPacket(this.sql));
 };
 
+Query.prototype.started = function() {
+  return this._started;
+};
+
+Query.prototype.ended = function() {
+  return this._ended;
+};
+
+Query.prototype.dequeue = function() {
+  if (this._started) {
+    throw new Error('Query already started.');
+  }
+  this.end();
```

Comment:
I'm not entirelry sure calling `this.end()` here won't mess something. I've tracked the code, it seems to behave as expected: it calls `this._callback` with no results, emits `end` which calls `Protocol._dequeue(this)`. But I'm not sure I covered all cases that may happen.